controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				spobj.Status.Phase = ocsv1.StorageProfilePhaseFailed
+				return []reconcile.Request{}
+			}
+
+			// Get the StorageCluster object
+			scList := &ocsv1.StorageClusterList{}
+			err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: obj.GetNamespace(), Limit: 1})
```

Comment:
Can we please use the abstraction `client` provides over ListOptions?

  ```suggestion
			err := r.Client.List(r.ctx, scList, &client.InNamespace(obj.GetNamespace()).Limit(1))
```

Reply:
This is the correct usage:
```suggestion
opts := &client.ListOptions{Limit: 1}
client.InNamespace(obj.GetNamespace()).ApplyToList(opts)
err := r.Client.List(r.ctx, scList, opts)
```

Reply:
I did make a mistake in my suggestion but you also don't need to run the `ApplyToList` call.
The clean way of using the client abstraction is to have bench one called and passed as a separate param

```suggestion
			err := r.Client.List(r.ctx, scList, &client.Inamespace(obj.GetNamespace()), client.Limit(1))
```